modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -268,46 +268,159 @@ struct IEUnit {
     }
 };
 
-struct IECallContext
+class IECallContext
 {
-    // Input parameters passed to an inference operation.
-    std::vector<cv::GArg> args;
-    cv::GShapes in_shapes;
+public:
+    IECallContext(const cv::GArgs                                   &  args,
+                  const std::vector<cv::gimpl::RcDesc>              &  outs,
+                  std::vector<cv::gimpl::GIslandExecutable::InObj>  && input_objs,
```

Comment:
Why isn't this `const`?

Reply:
Oh, it's a universal reference, I see

Reply:
So it isn't universal reference, input_objs/output_objs should be moved into the task context